filters/auth/authclient.go

```diff
@@ -91,6 +91,12 @@ func (ac *authClient) getTokenintrospect(token string, ctx filters.FilterContext
 	}
 	defer rsp.Body.Close()
 
+	// check 5xx status codes
+	if rsp.StatusCode >= 500 && rsp.StatusCode < 600 {
+		io.Copy(io.Discard, rsp.Body)
+		return nil, fmt.Errorf("token introspection failed with status code: %d", rsp.StatusCode)
+	}
```

Comment:
I think it is better to move this check into `if rsp.StatusCode != 200`

Reply:
> it is better

So that we have generic check that it is not 200 and then return specific error for 5xx.
I would also change `rsp.StatusCode < 600` to `rsp.StatusCode <= 599` - to highlight we are checking 5xx range.